src/millie.nim

```diff
@@ -10,46 +10,41 @@ const units = @[
   "P", # quadrillion
 ]
 
-## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
+
 proc millify*(
     value: int64,
     precision: int = 1,
     lowercase: bool = false,
     space: bool = false,
   ): string =
+  ## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
   
-  # validate if `vaue` can be parsed or not
-  var val: SomeNumber = parseNum(value)
+  # preallocate the resulting string
+  result = newStringOfCap(10)
 
-  # `prefix` will be used later when assembling
-  let prefix: string = if val < 0: "-" else: ""
+  var val = float(value)
 
+  # add a negative prefix if needed
+  if val < 0: result.add '-'
```

Comment:
Instead of using `strformat` and adding all strings together, I'm constructing the resulting string on-the-go, so we have no temporary strings.

Also Nim differentiates between `char` and `string`, and if we add a `char` it's a tiny bit better.